fastlane/lib/fastlane/helper/git_helper.rb

```diff
@@ -81,7 +81,7 @@ def self.last_git_commit_message
 
     # Returns the current git branch - can be replaced using the environment variable `GIT_BRANCH`
     def self.git_branch
-      return ENV['GIT_BRANCH'] if ENV['GIT_BRANCH'].to_s.length > 0 # set by Jenkins
+      return ENV['GIT_BRANCH'] if FastlaneCore::Env.truthy?('GIT_BRANCH') # set by Jenkins
```

Comment:
We can leave that as it was before or the semantics will change - e.g. if a user has a git branch called `false` this will fail the build on Jenkins.